File: framework/java/src/main/java/soda/unittest/job/codec/ObjectArrayCodec.java
package soda.unittest.job.codec;

import java.util.List;

import soda.unittest.DataUtils;

public class ObjectArrayCodec implements ICodec<List<?>, Object> {
	
	private Class<?> elementType;
	
	private int dimensions;
	
	public ObjectArrayCodec(Class<?> elementType, int D) {
		this.elementType = elementType;
		dimensions = D;
	}

	@Override
	public List<?> encode(Object object) {
		return (List<?>) DataUtils.toList(object);
	}

	@Override
	public Object decode(List<?> serial) {
		return DataUtils.toArray(serial, elementType, dimensions);
	}

}

File: framework/scala/src/main/scala/soda/scala/unittest/codec/ObjectArrayCodec.scala
package soda.scala.unittest.codec

import soda.unittest.DataUtils

class ObjectArrayCodec(elementType: Class[_], dimensions: Int) extends ICodec[Object, Object] {

    override def encode(obj: Object): Object = {
        DataUtils.toList(obj)
    }

    override def decode(ser: Object): Object = {
        DataUtils.toArray(ser, elementType, dimensions)
    }

}
